packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -168,9 +183,27 @@ - (instancetype)initWithCameraName:(NSString *)cameraName
   _motionManager = [[CMMotionManager alloc] init];
   [_motionManager startAccelerometerUpdates];
 
-  if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
-    return nil;
+  NSError *outError;
+  if ([_captureDevice lockForConfiguration:&outError]) {
+    [_videoCaptureSession beginConfiguration];
+
+    if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
+      [_videoCaptureSession commitConfiguration];
```

Comment:
why do we need to commit here if it fails? (I assume `![self setCpature...]` means it fails)

Reply:
There was crashes without `commitConfiguration`. At least other guys call it: https://github.com/xaphod/react-native-camera/blob/7ea241fa789072714121be8e7df791a672ba555c/ios/RN/RNCamera.m#L700

Reply:
in case of future regression, can you add a comment explaining this behavior?